hypothesis-python/src/hypothesis/strategies/_internal/strings.py

```diff
@@ -153,7 +153,7 @@ def __repr__(self):
     )
 
     def filter(self, condition):
-        if condition in (str.lower, str.title, str.upper):
+        if condition in self._nonempty_filters:
```

Comment:
No idea, but it seemed like this was the intent? Otherwise, `_nonempty_filters` is not used.

Reply:
Ah, this is tricky: here we're emitting a warning because it seems likely that you intended `str.islower` instead of `str.lower` (etc.); the `_nonempty_filters` are invoked via the call to `ListStrategy.filter` or `super().filter` below.

Adding a comment to explain that would be useful though!